system/jlib/jptree.cpp

```diff
@@ -7869,6 +7869,17 @@ jlib_decl StringBuffer & regenerateConfig(StringBuffer &jsonText, IPropertyTree
     return jsonText;
 }
 
+jlib_decl IPropertyTree * loadArgsIntoConfiguration(IPropertyTree *config, const char * * argv)
```

Comment:
Not sure if this is compatible with Gavin's recent changes. Also not sure why you are supplying some more command-line args later after the first call ?

Reply:
this is used by the slaves. They don't load a configuration file, but instead have the Thor configuration serialized to them, but they still have a least 1 command line parameter, which tell it what the --master is. In bare-metal case there are few other options e.g. --slavenum

Reply:
@richardkchapman 
In the container world is there any advantage to the slaves not accessing the same configuration files locally?
Also, we now have a global file which is loaded, and likely to be accessed by jlib to get at global information.  How does that fit in?  (And does this suggest that having a separate global file, rather than a section of the component file might not be the best approach?)

Reply:
There's no reason slaves should not see the same config file that the master does

I'm not sure it has any impact on global

Reply:
(note - there's already precendence for sharing config files- e.g. between agentexec and eclagent, or between eclccserver and the job it launches

Reply:
Currently thormaster/thorslave is a little different - in that the master loads the config, augements it with extra information, and serializes it to the slaves.

The slaves also need to pass different config options to the command line, in a container setup they only need --master= , in bare-metal they need to know a couple of other values, e.g. --slavenum

The slaves could instead see the same configuration file as the master, load it, and then the master could send only what additional config options it wants/needs to.
NB: I wouldn't be surprised if the properties the master sets now, could or should go away, either because they should already be set or defaulted before the master (or slave) sees them, or they just don't apply to container setup (e.g. @masterBuildTag).

Further changes like that should be left for a separate PR though.

What I needed now is a way to parse command line options (--xx), then slave starts, and to merge in the configuration being serialized from the master.